[lauhengyi] iP

lauhengyi opened the pull request:
### L.A.U.R.A

> "Location confirmed. The men who attacked [Stark Industries](https://marvelcinematicuniverse.fandom.com/wiki/Stark_Industries_Headquarters) are here." - J.A.R.V.I.S

LAURA helps you do stuff. It is:

- Capable
- Powerful
- Amazing

How does LAURA improve your life?

1. It offloads mental work from your brain
2. Gives you more time to think
3. You accomplish more stuff

Price: ~~$99.90/month~~ **FREE!!**

Milestones:

- [x] 0 users
- [x] 1 users
- [ ] 2-5 users
- [ ] 1000000000 users (coming soon!!)

#### L.A.U.R.A can even run code!!!!
Look at what it can run!
```java
System.out.println("Hello World!");
````
Laura is even smart enough to tell you that `Hello World!` will be printed on console before that code its even ran!! Its giving DeepSeek a run for its money! 😲😲

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

lauhengyi committed:
Renamed file, class and name

lauhengyi committed:
Added echoing

lauhengyi committed:
Added message class for messaging

lauhengyi committed:
Added list feature

lauhengyi committed:
Added task class

lauhengyi committed:
Added Task List

lauhengyi committed:
Added marking and unmarking

lauhengyi committed:
Added classes for different tasks

lauhengyi committed:
Added different tasks to task list

lauhengyi committed:
Added testing

lauhengyi committed:
Added exception cases

lauhengyi committed:
Updated tests

lauhengyi committed:
Added remove

lauhengyi committed:
Updated testcases

lauhengyi committed:
Handle save

lauhengyi committed:
Made encoding better

lauhengyi committed:
Added loading

lauhengyi committed:
Merge pull request #1 from lauhengyi/branch-Level-7

Branch level 7

lauhengyi committed:
Added UI Class

lauhengyi committed:
Added date parsing

lauhengyi committed:
Merge pull request #3 from lauhengyi/branch-Level-8

Branch level 8

lauhengyi committed:
Added parser

lauhengyi committed:
Made datapath less hardcoded

lauhengyi committed:
Add constructor + run method in Laura

lauhengyi committed:
Merge pull request #4 from lauhengyi/branch-A-MoreOOP

Branch a more oop

lauhengyi committed:
Added Packages

lauhengyi committed:
Merge pull request #5 from lauhengyi/branch-A-Packages

Added Packages

lauhengyi committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

lauhengyi committed:
Modify mainClass set

lauhengyi committed:
change name

lauhengyi committed:
Added date test

lauhengyi committed:
Added task test

lauhengyi committed:
Merge pull request #6 from lauhengyi/branch-A-JUnit

A j unit

lauhengyi committed:
Doc for Date

lauhengyi committed:
Added doc for Parser

lauhengyi committed:
Added Doc for TaskList

lauhengyi committed:
Add Doc for Exceptions and UI

lauhengyi committed:
Add docs for tasks

lauhengyi committed:
Merge pull request #7 from lauhengyi/A-JavaDoc

A java doc

lauhengyi committed:
Finish find method

lauhengyi committed:
Merge pull request #8 from lauhengyi/branch-Level-9

Finish find method

TobyCyan commented on src/main/java/laura/Date.java:
```diff
@@ -0,0 +1,44 @@
+public class Date {
+    private final LocalDate date;
+    /** The format that the date should be written in string */
+    private static final DateTimeFormatter stringFormat = DateTimeFormatter.ofPattern("dd/MM/yyyy");
```
I noticed that this could be a constant, perhaps use screaming camel case instead?
```suggestion
    private static final DateTimeFormatter STRING_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy");
```

TobyCyan commented on src/main/java/laura/Laura.java:
```diff
@@ -0,0 +1,29 @@
+        taskList.load();
+    }
+
+    public void run() {
+        UI.welcomeMessage();
+        Parser parser = new Parser(this.taskList);
+
+        while (!parser.hasEnded()) {
+            try {
+                parser.handleCommand();
+            } catch (LauraException e) {
+                UI.send(e.getMessage());
+            }
+        }
+    }
```
I like how you used intuitive naming for methods here, nice 👍

TobyCyan commented on src/main/java/laura/Parser.java:
```diff
@@ -0,0 +1,112 @@
+    /** Scanner object to scan for input */
+    private final Scanner scanner;
+    /** Whether the user has given the command to exit */
+    private boolean exit;
```
Maybe you can phrase this as a boolean instead?
```suggestion
    private boolean shouldExit;
```

TobyCyan commented on src/main/java/laura/Parser.java:
```diff
@@ -0,0 +1,112 @@
+            String description = details.substring(0, fI);
+            String timing = details.substring(fI + 7);
+            int tI = timing.indexOf(" /to ");
+            if (tI == -1) {
+                throw new LauraException("Event Task has no To value!");
+            }
+            String from = timing.substring(0, tI);
+            String to = timing.substring(tI + 5);
+
+            taskList.add(new EventTask(description, from, to));
+        } else if (input.startsWith("remove ")) {
+            int index;
+            try {
+                index = Integer.parseInt(input.substring(7));
+            } catch (NumberFormatException e) {
+                index = -1;
+            }
+            taskList.delete(index);
+        } else if (input.startsWith("mark ")) {
+            int index;
+            try {
+                index = Integer.parseInt(input.substring(5));
+            } catch (NumberFormatException e) {
+                index = -1;
+            }
+            taskList.mark(index);
+        } else if (input.startsWith("unmark ")) {
+            int index;
+            try {
+                index = Integer.parseInt(input.substring(7));
+            } catch (NumberFormatException e) {
+                index = -1;
+            }
+            taskList.unmark(index);
+        } else if (input.startsWith("find ")) {
+            String keyword = input.substring(5);
+            taskList.find(keyword);
+        } else {
+            throw new LauraException("Oops! I don't recognise this command!");
+        }
```
Perhaps you can isolate the first word out and use a switch case here?

TobyCyan commented on src/main/java/laura/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+public class TaskList {
+    /** Where the local data should be stored */
+    private final String dataPath;
+    /** The list that will hold the tasks*/
+    ArrayList<Task> list;
```
Not sure if I like the naming here, perhaps use tasks for a TaskList instead?
```suggestion
    /** The list that will hold the tasks*/
    ArrayList<Task> tasks;
```

TobyCyan commented on src/main/java/laura/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+    ArrayList<Task> list;
+
+    /**
+     * Create an instance of TaskList
```
Maybe use present tense for the first line here?
```suggestion
     * Creates an instance of TaskList.
```

TobyCyan commented on src/main/java/laura/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+        }
+        boolean isDone = args[1].equals("1");
+
+        return switch (args[0]) {
+            case "T" -> {
+                if (args.length != 3) {
+                    throw new DecodeException();
+                }
+                yield new ToDoTask(isDone, args[2]);
+            }
+            case "D" -> {
+                if (args.length != 4) {
+                    throw new DecodeException();
+                }
+                yield new DeadlineTask(isDone, args[2], args[3]);
+            }
+            case "E" -> {
+                if (args.length != 5) {
+                    throw new DecodeException();
+                }
+                yield new EventTask(isDone, args[2], args[3], args[4]);
+            }
+            default -> throw new DecodeException();
+        };
```
I like how you used switch case here! 👍

TobyCyan commented on src/main/java/laura/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+        }
+        Task removed = this.list.remove(index - 1);
+        this.save();
+        UI.send("Noted. I've removed this task:\n"
+                + removed +
+                "\nNow you have " + this.list.size() + " in this list.");
```
Perhaps the operator should be part of the line break here?
```suggestion
        UI.send("Noted. I've removed this task:\n"
                + removed 
                + "\nNow you have " + this.list.size() + " in this list.");
```

TobyCyan commented on src/main/java/laura/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+     * Filter the current TaskList to only Tasks which has the keyword
+     * @param keyword The keyword to filter by
+     */
+    public void find(String keyword) {
```
I like how you used short and simple method names here and in several other parts😄

TobyCyan commented on src/main/java/laura/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+     * @param keyword The keyword to filter by
+     */
+    public void find(String keyword) {
+        ArrayList<Task> filtered = new ArrayList<>();
```
Perhaps a more intuitive name here?
```suggestion
        ArrayList<Task> filteredTasks = new ArrayList<>();
```

TobyCyan commented on src/main/java/laura/UI.java:
```diff
@@ -0,0 +1,34 @@
+     */
+    public static void welcomeMessage() {
+        UI.send(
+                "Hello I'm Laura\n" +
+                        "What can I do for you?"
```
Similar to other parts, perhaps a line break before the + operator?
```suggestion
                "Hello I'm Laura\n"
                + "What can I do for you?"
```

TobyCyan commented on src/main/java/laura/UI.java:
```diff
@@ -0,0 +1,34 @@
+     * Sends a Message for the user to see
+     * @param message The Message to be sent
+     */
+    public static void send(String message) {
+        String line = "____________________________________________________________";
+        System.out.println("\n" + line);
+        System.out.println(message);
+        System.out.println(line);
+    }
```
Hey, I really like how you abstract the message format into a method here 😃

TobyCyan commented on src/main/java/laura/task/Task.java:
```diff
@@ -0,0 +1,69 @@
+     * @param keyword The keyword to be searched
+     * @return Whether the description has the keyword inside it
+     */
+    public boolean has(String keyword) {
```
I like how you include a query keyword here, but perhaps the naming could be a little bit clearer as such?
```suggestion
    public boolean hasKeyword(String keyword) {
```

swatirajesh277 replied:
Perhaps not the most intuitive method name here as it is a boolean method?

TobyCyan commented on src/test/java/laura/task/EventTaskTest.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    @Test
+    public void constructor_invalidDate_exceptionThrown() {
```
I like how you name your test cases, nice one 👍

TobyCyan reviewed (commented):
Overall, I found the naming convention of your code to be quite well done, and you have also closely followed the Java coding standards. However, I've noted down some minor coding standard violations here and there and some method/ variable names that I think could be made slightly more intuitive. There was a part where a series of very long if-else statements could be found, consider using switch case for better readability if that makes sense.

lauhengyi committed:
Update naming of Date field in src/main/java/laura/Date.java

Co-authored-by: Yong Jun Xi <89063663+TobyCyan@users.noreply.github.com>

lauhengyi committed:
fixed naming issue

lauhengyi committed:
Changed naming convention to shouldExit

lauhengyi committed:
Added checkstyle

lauhengyi committed:
Fixed style errors

lauhengyi committed:
Created initial basic template

lauhengyi committed:
Built basic ui

swatirajesh277 commented on src/main/java/laura/Date.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * @return The format of the date for data encoding
+     */
+    public String encode() {
```
I like how you have used an intuitive method name!

swatirajesh277 commented on src/main/java/laura/TaskList.java:
```diff
@@ -0,0 +1,201 @@
+     * @param keyword The keyword to filter by
+     */
+    public void find(String keyword) {
+        ArrayList<Task> filtered = new ArrayList<>();
```
Perhaps a more intuitive Array List name here?

swatirajesh277 commented on src/main/java/laura/TaskList.java:
```diff
@@ -0,0 +1,201 @@
+
+    @Override
+    public String toString() {
+        StringBuilder message = new StringBuilder();
```
I like the intuitive String name here! Nice job!

swatirajesh277 commented on src/test/java/laura/task/EventTaskTest.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    @Test
+    void unmark_success() throws LauraException{
```
Intuitive method name for tests! Very nice!

swatirajesh277 reviewed (commented):
Overall, your code is very good. The names of methods and variables follow the java coding standard properly. Nice work!

lauhengyi committed:
Made GUI

lauhengyi committed:
Merge pull request #9 from lauhengyi/branch-Level-10

Level 10

lauhengyi committed:
Fixed data bug

lauhengyi committed:
Add welcome message

lauhengyi committed:
Added assertions

Added assertions to check whether the file path indeeds leads to a file that is readable.

lauhengyi committed:
Rearranged code for better logical flow

lauhengyi committed:
Added final to ensure they are not modified

lauhengyi committed:
Added some javadoc

lauhengyi committed:
Add code comments

Come parsing code can be quite difficult to understand, commenting what various parts of the function does makes it clearer

lauhengyi committed:
Added better error handling for loaded errors

lauhengyi committed:
Merge pull request #10 from lauhengyi/branch-A-Assertions

Added assertions

lauhengyi committed:
Merge branch 'master' into branch-A-CodeQuality

lauhengyi committed:
Merge pull request #11 from lauhengyi/A-CodeQuality

A code quality

lauhengyi committed:
Added tag functionality to tasks

lauhengyi committed:
Made better handling for loading errors

lauhengyi committed:
Fixed bug with parser

lauhengyi committed:
Abstracted out functions in per command

lauhengyi committed:
Added tag operation in TaskList

lauhengyi committed:
Added tag command in parser

lauhengyi committed:
Merge pull request #12 from lauhengyi/branch-BCD-Extension

Branch bcd extension

lauhengyi committed:
Added CSS, removed print statements

lauhengyi committed:
Added resizing capabilities

lauhengyi committed:
Added asymmetric GUI

lauhengyi committed:
Merge pull request #14 from lauhengyi/branch-A-BetterGui

A better gui

lauhengyi committed:
Tweaked some colors

lauhengyi committed:
Fixed test cases, and added test cases for tag

lauhengyi committed:
Merge pull request #15 from lauhengyi/branch-A-MoreTesting

Fixed test cases, and added test cases for tag

lauhengyi committed:
Added credit

lauhengyi committed:
Added title window + added Better welcome message

lauhengyi committed:
Added documentation

lauhengyi committed:
Merge pull request #16 from lauhengyi/branch-A-UserGuide

Added documentation

lauhengyi committed:
Fixed bug with missing parent directory

lauhengyi committed:
Remove comment

lauhengyi committed:
Add comment to Data.java

lauhengyi committed:
Remove comment to Data.java

lauhengyi committed:
Add comment to Data.java

lauhengyi committed:
Remove comment to Data.java

lauhengyi committed:
Add comment to Data.java
